gateway/tests/tests/tests.rs

```diff
@@ -118,16 +119,41 @@ async fn test_gateway_authentication() -> Result<()> {
     task_group.shutdown_join_all().await
 }
 
+/// Test that a given endpoint/functionality of func fails with the wrong password but works with the correct one
 async fn test_auth<Fut>(gw_password: &str, func: impl Fn(String) -> Fut) -> Result<()>
 where
     Fut: Future<Output = Result<Response, Error>>,
 {
     assert_eq!(
-        // use random password here
-        func("foobar123456789".to_string()).await?.status(),
+        retry(
+            "fn".to_string(),
+            || async {
+                func(format!("foobar{}", gw_password))
+                    .await
+                    .map_err(|e| anyhow::anyhow!(e))
+            },
+            Duration::from_secs(1),
+            3,
+        )
+        .await?
+        .status(),
+        401
+    );
+    assert_ne!(
+        retry(
+            "fn".to_string(),
```

Comment:
`"fn"`? :D . It's OK. But weird. :D

Reply:
:) I could label these functions appropriately by passing a context param to `test_auth`, but we generally don't care about the label. Might only come in handy when debugging the test suite. Opted not to bother with it now